src/remote/remote.ts

```diff
@@ -733,6 +765,95 @@ export class Remote {
 		return ` --log-dir ${escapeCommandArg(logDir)} -v`;
 	}
 
+	/**
+	 * Waits for an agent condition with progress notification and error handling.
+	 * Returns the updated agent or undefined if the user chooses to close remote.
+	 */
+	private async waitForAgentWithProgress(
+		monitor: WorkspaceMonitor,
+		agent: WorkspaceAgent,
+		options: {
+			progressTitle: string;
+			timeoutMs: number;
+			errorDialogTitle: string;
+		},
+		checker: (agent: WorkspaceAgent) => WorkspaceAgent | undefined,
+	): Promise<WorkspaceAgent | undefined> {
+		try {
+			const foundAgent = await vscode.window.withProgress(
+				{
+					title: options.progressTitle,
+					location: vscode.ProgressLocation.Notification,
+				},
+				async () =>
+					this.waitForAgentCondition(
+						monitor,
+						agent,
+						checker,
+						options.timeoutMs,
+						`Timeout: ${options.errorDialogTitle}`,
+					),
+			);
+			return foundAgent;
+		} catch (error) {
+			this.logger.error(options.errorDialogTitle, error);
+			const result = await this.vscodeProposed.window.showErrorMessage(
+				options.errorDialogTitle,
+				{
+					useCustom: true,
+					modal: true,
+					detail: error instanceof Error ? error.message : String(error),
+				},
+				"Close Remote",
+			);
+			if (result === "Close Remote") {
+				await this.closeRemote();
+			}
```

Comment:
Error dialogs can be closed without pressing the button I think, right?  Should we always close the remote?  I think there is no scenario in the case of an error where we would want to keep the remote open.

Actually, could we let this bubble up?  We close the remote on errors from setting up the remote already.  Kinda nice to have one place where we catch an error and display it.